### ISSUE TITLE
[BUG] Net::SMTP#start signature differs depending on Ruby's version

### ISSUE THREAD

**Issue by evserykh** [truemail-rb/truemail#188]
### New bug checklist

- [x] I have updated `truemail` to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/truemail-rb/truemail/blob/master/CONTRIBUTING.md)
- [x] I have read the [documentation](https://truemail-rb.org/truemail-gem)
- [x] I have searched for [existing GitHub issues](https://github.com/truemail-rb/truemail/issues)

### Bug description

I'm using the following Ruby version `ruby 2.5.5p157 (2019-03-15 revision 67260) [x86_64-linux]`

Here is the code I use:
```
require 'net/smtp'
require 'truemail'

session = Net::SMTP.new('gmail-smtp-in.l.google.com', 25)
session.start('example.com') do |smtp|
  smtp.mailfrom 'test@example.com'
  p smtp.rcptto 'foobar@gmail.com'
end

Truemail.configure do |config|
  config.verifier_email = 'test@example.com'
  config.smtp_fail_fast = true
end

jj JSON.parse(Truemail.validate('foobar@gmail.com').as_json)
```

and here is the output:
```
#<Net::SMTP::Response:0x000055af65823998 @status="250", @string="250 2.1.5 OK n1si4121035lji.231 - gsmtp\n">
{
  "date": "2021-12-03 09:56:05 +0000",
  "email": "foobar@gmail.com",
  "validation_type": "smtp",
  "success": false,
  "errors": {
    "smtp": "smtp error"
  },
  "smtp_debug": [
    {
      "mail_host": "64.233.164.26",
      "port_opened": true,
      "connection": false,
      "errors": {
        "connection": "SMTP-AUTH requested but missing secret phrase"
      }
    }
  ],
  "configuration": {
    "validation_type_by_domain": null,
    "whitelist_validation": false,
    "whitelisted_domains": null,
    "blacklisted_domains": null,
    "blacklisted_mx_ip_addresses": null,
    "dns": null,
    "not_rfc_mx_lookup_flow": false,
    "smtp_fail_fast": true,
    "smtp_safe_check": false,
    "email_pattern": "default gem value",
    "smtp_error_body_pattern": "default gem value"
  }
}
```
After some research I figured out before Ruby 3 the `Net::SMTP#start` method's definition looks like this:
```
def start(helo = 'localhost',
          user = nil, secret = nil, authtype = nil)
```
and starting from Ruby 3 the method's definition has been changed to this:
```
def start(*args, helo: nil,
          user: nil, secret: nil, password: nil, authtype: nil, tls_verify: true, tls_hostname: nil)
```

This monkey patch allows to fix my case:
```
module Truemail
  module Validate
    class Smtp
      class Request
        def start_session_wrapper(&block)
          ruby3 = ::RUBY_VERSION[/\A3\..+\z/]
          if ruby3
            session.start(configuration.verifier_domain, tls_verify: false, &block)
          else
            session.start(configuration.verifier_domain, &block)
          end
        end

        def run
          start_session_wrapper do |smtp_request|
            response.connection = response.helo = true
            smtp_handshakes(smtp_request, response)
          end
        rescue => error
          retry if attempts_exist?
          assign_error(attribute: :connection, message: compose_from(error))
        end
      end
    end
  end
end
``` 

So if I don't miss anything I could create a PR using this patch 🤷‍♂️

**Comment by bestwebua** [comment]
Hi, @evserykh! What you truemail version, show your `Truemail::VERSION` output please. Which version are you using? This issue has been fixed in latest release: https://github.com/truemail-rb/truemail/pull/186

**Comment by bestwebua** [comment]
Oh, my fault. Latest updates have broke `Net::SMTP#start` Ruby 2.x compatibility. Looks like that I should as soon as possible write `truemail` integration tests using [smtpmock](https://github.com/mocktools/go-smtp-mock) under the hood to get rid bugs like this one in future 👯‍♀️

**Comment by bestwebua** [comment]
@evserykh, this fix is available in latest (2.5.4) truemail release. Cheers!
